internal/pkg/types/v1alpha1/build.go

```diff
@@ -0,0 +1,216 @@
+package v1alpha1
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/hashicorp/go-multierror"
+	"golang.org/x/sys/unix"
+	"golang.org/x/xerrors"
+)
+
+func (p *Pkg) Build() error {
+	if err := environment(); err != nil {
+		return err
+	}
+
+	println("Build:", os.Getenv("BUILD"))
+	println("Host:", os.Getenv("HOST"))
+	println("Target:", os.Getenv("TARGET"))
+
+	for _, step := range p.Steps {
+		if _, err := os.Stat("/tmp"); err != nil {
+			if os.IsNotExist(err) {
+				// TODO(andrewrynhard): Remove this directory.
+				log.Println("Creating /tmp")
+				os.MkdirAll("/tmp", 0777)
+			}
+		}
+		dir, err := ioutil.TempDir("", "pkg")
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer os.RemoveAll(dir)
+
+		if err := os.Chdir(dir); err != nil {
+			log.Fatal(err)
+		}
+		if err := step.download(dir); err != nil {
+			log.Fatal(err)
+		}
+		if err := step.prepare(p.Shell); err != nil {
+			log.Fatal(err)
+		}
+		if err := step.build(p.Shell); err != nil {
+			log.Fatal(err)
+		}
+		if err := step.install(p.Shell); err != nil {
+			log.Fatal(err)
+		}
+		if err := step.test(p.Shell); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	return nil
+}
+
+func environment() error {
```

Comment:
this func is probably fine for now, but we should look at either setting these as defaults that can be overridden or moved to bldr.yaml